Complete docs (#137)
spacetelescope/cosmo: docs/source/sms.rst

SMS Data Ingestion
==================
Some of the monitors in COSMO rely on data that are found in Science Mission Schedule (SMS) reports.
These reports are produced outside of the COS Branch and supplied via *txt* files.

A variety of data on exposures is provided in these report files, and in some cases can actually be used to verify
information in the regular data products themselves.

SMS File Format and Naming Conventions
--------------------------------------
The Report files use a human readable table format, which requires special processing to read since it's not a "typical"
table format (such as *csv*).

The naming convention for these files are comprised of two different ID codes:
 1. A "SMS ID" which uniquely identifies a particular SMS report
 2. A "Version ID" which identifies the version of a particular SMS

> The SMS ID is based on the date that the report was created, and is the first six digits in the report filename.

Reviewer: replace "is based on the date that the report was created" with "is the start data of the SMS in the form YYDDD"

> The SMS ID is based on the date that the report was created, and is the first six digits in the report filename.

Author: ```suggestion
The SMS ID is the start data of the SMS in the form YYDDD, and is the first six digits in the report filename.
```

For example:

.. code-block:: python

    import pandas as pd
    from cosmo.sms import SMSFileStats, SMSTable

    # select all sms records
    query = SMSFileStats.select()  # Returns a query object

     # Put the data in a pandas DataFrame
    df = pd.DataFrame(list(query.dicts()))

    # Show data records from a single SMS report using the SMSID
    report = SMSFileStats.get(SMSFileStats.SMSID == '123456')

    # Or
    report = SMSFileStats.get('123456')  # The SMSID is the primary key for SMSFileStats, so this syntax also works

    # Or
    report = SMSFileStats['123456']

    report.exposures  # returns a peewee.Query object

    list(report.exposures.dicts())  # returns a list of dictionaries of the table data

    # Query the SMS Data
    all_data = SMSTable.select()

    # Query and filter SMS Data
    fuv_data = SMSTable.select().where(SMSTable.DETECTOR == 'FUV')

SMS Ingestion Rules
-------------------
SMS reports are ingested into the local SMS database, ``sms.db`` using a set of rules to ensure that the correct
information is stored.

Ingestion rules are as follows:

>  - If multiple versions of the same SMS is available, the most recent version is used (largest alpha-numeric Version ID).

Reviewer: is -> are

 - If a new version of an SMS is found with a larger Version ID, that new report will supersede the current entry for
   that particular SMS.
 - If data (identified by the exposure key) from one SMS is also found in another SMS, the data will be ingested from
   the SMS report with the larger SMS ID (and therefore more recent), and will replace any existing matching records.

.. note::

    These rules ensure that the data in SMSTable is accurate, but can also result in "stale" reports in SMSFileStats if
    data occurs in a more recent SMS report.

    It's also possible to have SMSTable records that don't actually correspond to any COS dataset.
    This is due to the fact that the SMS reports reflect the observation schedule, which can change a number of times
    right up to the planned observation date.
    As this happens, ROOTNAMEs and other exposure information is updated accordingly, and since the ROOTNAME is based on

>     an incrementing naming system, exposures can be assigned several ROOTNAMEs until the "final" schedule is established.

Author: ```suggestion
    an incrementing naming system, exposures can be assigned several ROOTNAMEs until the "final" schedule is established.
    Additionally, if an exposure does not execute (for example, when COS enters safe mode), those exposures will still
    appear in the SMS.
```

>     an incrementing naming system, exposures can be assigned several ROOTNAMEs until the "final" schedule is established.

Reviewer: Could also note that if the exposures don't execute for some reason (e.g. instrument goes into safe mode), those exposures will still appear in SMS.
